readthedocs/vcs_support/backends/git.py

```diff
@@ -174,11 +174,7 @@ def fetch(self):
         code, stdout, stderr = self.run(*cmd)
         return code, stdout, stderr
 
-    def checkout_revision(self, revision=None):
-        if not revision:
-            branch = self.default_branch or self.fallback_branch
-            revision = 'origin/%s' % branch
-
+    def checkout_revision(self, revision):
```

Comment:
Do we need to change the value passed in here? Seems we're also depending on `default_branch` and `fallback_branch` here:

https://github.com/readthedocs/readthedocs.org/blob/239d8905896d9bc9a7ec118b231a9c315f40f43c/readthedocs/vcs_support/backends/git.py#L326-L327

Reply:
I think you are reading old code that is deleted/adapted by this PR. In particular, to remove the usage of `default_branch` and `fallback_branch` 😄 

This is what I'm seeing this in the current PR at the location you shared: https://github.com/readthedocs/readthedocs.org/blob/5fc00a3826e2f52b7246bd7eff80f349ff0789bc/readthedocs/vcs_support/backends/git.py#L340-L342

Let me know if I'm reading you comment correctly.

Reply:
Heh yea, probably 🙃